### ISSUE TITLE
assertPostTypeTerms comparing stale cached terms 

### ISSUE THREAD

**Issue by davidbenton** [stephenharris/WordPressBehatExtension#17]
I have a scenario that tests bulk editing of a custom post type and custom taxonomy. My first bulk edit works, and this step passes:
```
Then the document "Multilingual Doc" should have document_tag terms "connections"
```
But bulk editing the same document again, the expected result is compared against the same `$actual_terms` as the first time, even though an additional term has been added to the post (verified through screenshot and manually visiting the site), leading to a failed test.

**Comment by davidbenton** [comment]
[WordPressPostTrait.php](https://github.com/stephenharris/WordPressBehatExtension/blob/e4370ec6e7e1904bfd6d0f78104b4d3358ec807b/src/Context/PostTypes/WordPressPostTrait.php#L103) has a line `clean_post_cache($post->ID);` which is presumably to avoid this. Replacing that with the more aggressive `wp_cache_flush();` solves this issue for me. Any concerns about that as fix?

**Comment by stephenharris** [comment]
`clean_post_cache` calls `clean_object_term_cache` which is what cache the terms of the post. So `clean_post_cache` is the correct function to call here.

Each taxonomy is cached independently, so it iterates over `get_object_taxonomies()` to clear all the taxonomies. Could it be that the taxonomy you are using is not returned in that function call?

Are you using a custom taxonomy?

**Comment by davidbenton** [comment]
Yes, it is a custom taxonomy. `get_object_taxonomies($post)` called from within `assertPostTypeTerms` returns that custom taxonomy and the other expected taxes for that post type.

**Comment by stephenharris** [comment]
Ah, ok `get_the_terms()` uses `wp_get_object_terms()` which in turn uses `get_terms()` and the query is also cached, and not cleared by `clean_post_cache()`. Just to confirm that this is the case, could you try replacing `get_the_terms( ... )` with the following:

    wp_get_object_terms( $post->ID, $taxonomy, array(
        'cache_doman' => uniqid(rand(), true)
    ) );

**Comment by davidbenton** [comment]
I get the same (stale cached) output from that as from `get_the_terms( ... )`.
